### PR TITLE
Add Dockerfile

### PR DESCRIPTION
**Pull request by dzaytsev91**
## Description

Added docker image and update readme to run program with docker.

Fixes https://github.com/github/linguist/issues/46

## Checklist:

- [x] **I am adding new or changing current functionality**
  
  - [x] I have added or updated the tests for the new or changed functionality.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -119,7 +119,15 @@ github-linguist.gemspec
 lib/linguist.rb
 …
 ```
-
+## Docker
+If you have docker installed you can build an image and run this as a container.
+```
+docker build -t my-linguist .
```
Comment (pchaigno) [inline README.md]: I think this should just be `linguist` in the example. It's not like their implementing their own child version of Linguist.
```suggestion
docker build -t linguist .
```

File: README.md [hunk 2]
```diff
@@ -119,7 +119,15 @@ github-linguist.gemspec
...
+```
+Move to project folder then run.
+```
+docker run --rm -v `pwd`:`pwd` -w `pwd` -t my-linguist
```
Comment (pchaigno) [inline README.md]: This is neat!
We might want to give an example for `breakdown` too.
```suggestion
docker run --rm -v `pwd`:`pwd` -w `pwd` -t linguist
docker run --rm -v `pwd`:`pwd` -w `pwd` -t linguist github-linguist --breakdown
```

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,12 @@
+FROM ruby:alpine
+
+RUN apk --update add --virtual build_deps \
+    build-base \
+    libc-dev \
+    linux-headers \
+    cmake \
+    && apk --no-cache add icu-dev libressl-dev \
+    && gem install github-linguist \
+    && apk del build_deps
```
Comment (pchaigno) [inline Dockerfile]: Any reason to not uninstall the other packages? It looks like we don't need them anymore and it would make the image smaller.

Comment (dzaytsev91) [reply Dockerfile]: Good point, removed useless

Comment (pchaigno) [reply Dockerfile]: I think you need those packages to build Linguist (at least `build-essential`). I was just suggesting we remove them after the installation.

**Comment by pchaigno** [comment]
@lildude What would be the proper way to test with GitHub Actions?

File: Dockerfile (hunk 3 above)
Comment (dzaytsev91) [reply Dockerfile]: Fixed

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,12 @@
+FROM ruby:alpine
+
+RUN apk --update add --virtual build_deps \
+    build-base \
+    libc-dev \
+    linux-headers \
+    cmake \
+    && apk --no-cache add icu-dev libressl-dev \
+    && gem install github-linguist \
+    && apk del build_deps build-base libc-dev linux-headers
```
Comment (pchaigno) [inline Dockerfile]: What about removing cmake, icu-dev, and libressl-dev?

Comment (dzaytsev91) [reply Dockerfile]: Removed `cmake`, but removing `icu-dev` or `libressl-dev` causes errors when trying to run  `linguist`
<details>
Traceback (most recent call last):
	16: from /usr/local/bundle/bin/github-linguist:23:in `<main>'
	15: from /usr/local/bundle/bin/github-linguist:23:in `load'
	14: from /usr/local/bundle/gems/github-linguist-7.6.1/bin/github-linguist:5:in `<top (required)>'
	13: from /usr/local/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb:54:in `require'
	12: from /usr/local/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb:54:in `require'
	11: from /usr/local/bundle/gems/github-linguist-7.6.1/lib/linguist.rb:6:in `<top (required)>'
	10: from /usr/local/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb:54:in `require'
	 9: from /usr/local/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb:54:in `require'
	 8: from /usr/local/bundle/gems/github-linguist-7.6.1/lib/linguist/repository.rb:1:in `<top (required)>'
	 7: from /usr/local/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb:54:in `require'
	 6: from /usr/local/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb:54:in `require'
	 5: from /usr/local/bundle/gems/github-linguist-7.6.1/lib/linguist/lazy_blob.rb:3:in `<top (required)>'
	 4: from /usr/local/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb:54:in `require'
	 3: from /usr/local/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb:54:in `require'
	 2: from /usr/local/bundle/gems/rugged-0.28.3.1/lib/rugged.rb:8:in `<top (required)>'
	 1: from /usr/local/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb:54:in `require'
/usr/local/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb:54:in `require': cannot load such file -- rugged/2.6/rugged (LoadError)
	17: from /usr/local/bundle/bin/github-linguist:23:in `<main>'
	16: from /usr/local/bundle/bin/github-linguist:23:in `load'
	15: from /usr/local/bundle/gems/github-linguist-7.6.1/bin/github-linguist:5:in `<top (required)>'
	14: from /usr/local/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb:54:in `require'
	13: from /usr/local/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb:54:in `require'
	12: from /usr/local/bundle/gems/github-linguist-7.6.1/lib/linguist.rb:6:in `<top (required)>'
	11: from /usr/local/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb:54:in `require'
	10: from /usr/local/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb:54:in `require'
	 9: from /usr/local/bundle/gems/github-linguist-7.6.1/lib/linguist/repository.rb:1:in `<top (required)>'
	 8: from /usr/local/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb:54:in `require'
	 7: from /usr/local/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb:54:in `require'
	 6: from /usr/local/bundle/gems/github-linguist-7.6.1/lib/linguist/lazy_blob.rb:3:in `<top (required)>'
	 5: from /usr/local/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb:54:in `require'
	 4: from /usr/local/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb:54:in `require'
	 3: from /usr/local/bundle/gems/rugged-0.28.3.1/lib/rugged.rb:6:in `<top (required)>'
	 2: from /usr/local/bundle/gems/rugged-0.28.3.1/lib/rugged.rb:10:in `rescue in <top (required)>'
	 1: from /usr/local/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb:54:in `require'
/usr/local/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb:54:in `require': Error loading shared library libssl.so.45: No such file or directory (needed by /usr/local/bundle/gems/rugged-0.28.3.1/lib/rugged/rugged.so) - /usr/local/bundle/gems/rugged-0.28.3.1/lib/rugged/rugged.so (LoadError)
</details>

Comment (pchaigno) [reply Dockerfile]: Ah. Weird, I didn't get the same errors on my setup.
Let's leave them then!

**Comment by pchaigno** [comment]
> @lildude What would be the proper way to test with GitHub Actions?

@lildude Nudge :-)

**Comment by dzaytsev91** [comment]
@lildude :)

**Comment by smola** [comment]
For GitHub Actions, you could start with a new job that executes `docker build .` and then smoke test it by doing `docker run ...` on a single file or the linguist repo to check that it doesn't explode.

**Comment by pchaigno** [comment]
@dzaytsev91 Are you still working on this? Any chance you could add a basic GitHub Action, as @smola described, to simply build and run the Docker image on the Linguist repo? I think you'll just need to checkout the code, as illustrated in [this example from the docs](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#example-using-action-in-the-same-repository-as-the-workflow), then run the two `build` and `run` commands.

**Comment by dzaytsev91** [comment]
@pchaigno Sure, I'll try to add it in a week

**Comment by lildude** [comment]
Whooops. Lost this is my massive inbox. @smola's suggestion sounds good to me.

**Comment by lildude** [comment]
@dzaytsev91 Do you mind if I push the necessary tests to this PR?

**Comment by dzaytsev91** [comment]
@lildude Of course not

**Comment by lildude** [comment]
I've added a test and tweaked the wording in the README. How does this look peeps?

**Review by Alhadis — APPROVED** [review]

**Comment by lildude** [comment]
@pchaigno I think all your concerns have been addressed. Happy with this now?

**Review by pchaigno — APPROVED** [review]
LGTM!

**Pull request merged**